lib/msal-common/test/request/RequestParameterBuilder.spec.ts

```diff
@@ -407,14 +416,92 @@ describe("RequestParameterBuilder unit tests", () => {
         ).toBe(true);
     });
 
-    it("doesn't add client assertion and client assertion type if they are empty strings", () => {
-        const clientAssertion = {
+    it("doesn't add client assertion (string) and client assertion type if they are empty strings", async () => {
+        const clientAssertion: ClientAssertion = {
             assertion: "",
             assertionType: "",
         };
 
         const requestParameterBuilder = new RequestParameterBuilder();
-        requestParameterBuilder.addClientAssertion(clientAssertion.assertion);
+        requestParameterBuilder.addClientAssertion(
+            await getClientAssertion(
+                clientAssertion.assertion,
+                "client_id",
+                "optional_token_endpoint"
+            )
+        );
+        requestParameterBuilder.addClientAssertionType(
+            clientAssertion.assertionType
+        );
+        const requestQueryString = requestParameterBuilder.createQueryString();
+        expect(
+            requestQueryString.includes(AADServerParamKeys.CLIENT_ASSERTION)
+        ).toBe(false);
+        expect(
+            requestQueryString.includes(
+                AADServerParamKeys.CLIENT_ASSERTION_TYPE
+            )
+        ).toBe(false);
+    });
+
+    it("adds clientAssertion (ClientAssertionCallbackFunction) and assertionType if they are provided by the developer", async () => {
+        const clientAssertionCallbackFunction: ClientAssertionCallbackFunction =
+            (_config: ClientAssertionConfig) => {
+                return Promise.resolve("testAssertion");
```

Comment:
You can assert that _config is populated with clientId and tokenEndpoint here.

Reply:
I'll be updating the unit tests shortly. I wanted to see if you approved of the proof-of-concept in this PR.